src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisWebhookService.java

```diff
@@ -0,0 +1,102 @@
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.PyrisPipelineExecutionSettingsDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.lectureIngestionWebhook.PyrisLectureUnitWebhookDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.lectureIngestionWebhook.PyrisWebhookLectureIngestionExecutionDTO;
+import de.tum.in.www1.artemis.service.connectors.pyris.job.IngestionWebhookJob;
+
+@Service
+@Profile("iris")
+public class PyrisWebhookService {
+
+    private static final Logger log = LoggerFactory.getLogger(PyrisWebhookService.class);
+
+    private final PyrisConnectorService pyrisConnectorService;
+
+    private final PyrisJobService pyrisJobService;
+
+    @Value("${server.url}")
+    private String artemisBaseUrl;
+
+    public PyrisWebhookService(PyrisConnectorService pyrisConnectorService, PyrisJobService pyrisJobService) {
+        this.pyrisConnectorService = pyrisConnectorService;
+        this.pyrisJobService = pyrisJobService;
+    }
+
+    private String attachmentToBase64(AttachmentUnit attachmentUnit) {
+        Path path = FilePathService.actualPathForPublicPathOrThrow(URI.create(attachmentUnit.getAttachment().getLink()));
+        // Path path = Path.of(Optional.ofNullable(attachment).map(Attachment::getLink).orElse(""));
+        try {
+            byte[] fileBytes = Files.readAllBytes(path);
+
+            return Base64.getEncoder().encodeToString(fileBytes);
+        }
+        catch (IOException e) {
+            return e.getMessage();
+        }
+    }
+
+    private PyrisLectureUnitWebhookDTO processAttachments(Boolean toUpdate, AttachmentUnit attachmentUnit) {
+        try {
+            if (toUpdate) {
+                int lectureUnitId = attachmentUnit.hashCode();
+                String lectureUnitName = attachmentUnit.getName();
+                int lectureId = attachmentUnit.getLecture().hashCode();
+                String lectureTitle = attachmentUnit.getLecture().getTitle();
+                int courseId = attachmentUnit.getLecture().getCourse().hashCode();
+                String courseTitle = attachmentUnit.getLecture().getCourse().getTitle();
+                String courseDescription = attachmentUnit.getLecture().getCourse().getDescription() == null ? "" : attachmentUnit.getLecture().getCourse().getDescription();
+                String base64EncodedPdf = attachmentToBase64(attachmentUnit);
+                return new PyrisLectureUnitWebhookDTO(true, base64EncodedPdf, lectureUnitId, lectureUnitName, lectureId, lectureTitle, courseId, courseTitle, courseDescription);
+            }
+            else {
+                int lectureUnitId = attachmentUnit.hashCode();
+                int lectureId = attachmentUnit.getLecture().hashCode();
+                int courseId = attachmentUnit.getLecture().getCourse().hashCode();
+                return new PyrisLectureUnitWebhookDTO(false, "", lectureUnitId, "", lectureId, "", courseId, "", "");
```

Comment:
What is the purpose of this if it passes no actual data to Pyris?

Reply:
This is for deletion purpose we don#t need the data and the names. We only need the ids to delete from the database.

Reply:
Then I would maybe rename the `toUpdate` parameter to say `shouldDelete` or smth. Or even better just add a separate method for this.

Reply:
I would just rename it to shouldUpdate for now.